akd_quorum/src/node/mod.rs

```diff
@@ -0,0 +1,2217 @@
+                            self.handle_reception_timeout().await?
+                        }
+                        Err(other_err) => {
+                            // comms channel errors should bubble up since that signifies a bigger issue that a restart may be necessary for
+                            return Err(QuorumOperationError::Communication(other_err));
+                        }
+                        Ok(received) => {
+                            // deserialization or message handling errors should not require a node reboot.
+                            match self.decrypt_message(received.message.clone()).await {
+                                Ok(message) => {
+                                    // move clone of state & self, + received into the async context & spawn onto green thread
+                                    // to free up message reception
+                                    let self_clone = self.clone();
+                                    let _ = tokio::spawn(async move {
+                                        if let Err(err) = self_clone
+                                        .handle_inter_node_message_helper(
+                                            message,
+                                            received.timeout,
+                                            received.reply,
+                                        )
+                                        .await {
+                                            warn!("Error handling message: {}", err);
+                                            // TODO: other logs or stat counters?
+                                        }
+                                    });
+                                }
+                                Err(err) => {
+                                    error!("Error decrypting node message");
+                                    return Err(err);
+                                }
+                            }
+                        }
+                    }
+                },
+                public_message = receiver.recv() => {
+                    if let Some(msg) = public_message {
+                        self.handle_message(NodeMessage::<H>::Public(msg)).await?;
+                    } else {
+                        warn!("Message receive handler received and empty message on the public message receive port");
+                    }
+                }
+            }
+        }
+    }
+
+    async fn handle_reception_timeout(&self) -> Result<(), QuorumOperationError> {
+        // Message reception timeout
+
+        // Perform a "state handler timer tick"
+        let node_id = self.state.config.read().await.node_id;
+        let _ = self
+            .handle_message(NodeMessage::Internal(
+                node_id,
+                InterNodeMessage::<H>::TimerTick,
+            ))
+            .await?;
+
+        Ok(())
+    }
+
+    async fn handle_inter_node_message(
```

Comment:
I'd suggest to rename `handle_inter_node_message` into something like `process_inter_node_message` to not confuse with others `inter_node_message_handler` and `handle_inter_node_message_helper`